app/src/main/java/com/checkin/app/checkin/auth/activities/AuthenticationActivity.kt

```diff
@@ -0,0 +1,290 @@
+                .build()
+        val signInIntent = GoogleSignIn.getClient(this, gso).signInIntent
+        startActivityForResult(signInIntent, RC_AUTH_GOOGLE)
+    }
+
+    override fun onFacebookAuth(loginResult: LoginResult) {
+        showProgress()
+        val accessToken = loginResult.accessToken
+        authViewModel.setProviderIdToken(accessToken.token)
+        val credential = FacebookAuthProvider.getCredential(accessToken.token)
+        authenticateWithCredential(credential)
+    }
+
+    override fun onPhoneAuth(phoneNo: String) {
+        val action = AuthOptionFragmentDirections.actionVerifyOtp(phoneNo)
+        navController.navigate(action)
+    }
+
+    private fun successAuth(data: AuthResultModel) {
+        Handler().post {
+            val authToken = data.token
+            val account = Account(resources.getString(R.string.app_name), Constants.ACCOUNT_TYPE)
+            val accountManager = AccountManager.get(this)
+            val userData = Bundle()
+            userData.putLong(Constants.ACCOUNT_UID, data.userId)
+            accountManager.addAccountExplicitly(account, null, userData)
+            accountManager.setAuthToken(account, AccountManager.KEY_AUTHTOKEN, authToken)
+            startService(Intent(applicationContext, DeviceTokenService::class.java))
+            userViewModel.userData.observe(this, Observer {
+                it?.let {
+                    when (it.status) {
+                        Resource.Status.SUCCESS -> {
+                            if (it.data?.firstName == defaultFirstName) {
+                                fetchUserDetails(name = null)
+                                //so if the firebase has  a name then that name would be the default name and the app will next time open with the
+                                //name firebase gave otherwise if the username is Checkin(default name) then the case would be that the name would be checked
+                                //if so that the name is checkin then the details screen would be fetched
+                            } else {
+                                openHome()
+                            }
+                        }
+                    }
+                }
+            })
+            userViewModel.fetchUserData()
+        }
+    }
+
+    private fun openHome() {
+        startActivity(Intent(this, HomeActivity::class.java))
+        finish()
+    }
+
+    override fun onSuccessVerification(credential: PhoneAuthCredential, idToken: String, phone: String) {
+        authViewModel.setFireBaseIdToken(idToken)
+        phoneNo = phone
+        phoneToken = idToken
+        authViewModel.login()
+    }
+
+    override fun onCancelVerification() {
+
+    }
+
+    override fun onFailedVerification(exception: Exception) {
+        toast(exception.message ?: getString(R.string.error_authentication_phone))
+        if (exception !is InvalidOTPException) {
+            toast("Invalid OTP try Again")
```

Comment:
you're showing two toast one after another here... bad UX
Use proper if-else and also log exceptions (Use Utils.log for it)